File: agent.py
from a_star import aStar


class Agent:

    def __init__(self, start, cur, goal, name=None, path=None):
        self.start = start
        self.current = cur
        self.goal = goal
        self.name = name
        self.path = path

    def set_path(self, path):
        self.path = path

    def set_astar_path(self, new_maze, potential_map, type):
        if type == 'renew':
            self.path = aStar(potential_map, self.current, self.goal, new_maze)
        elif type == 'init':
            self.path = aStar(potential_map, self.start, self.goal, new_maze)

    def move_path(self):
        if len(self.path) != 2:
            print(len(self.path))
            print(self.current)
            self.current = self.path[-2]
            return True
        return False

File: C++/Agent.cpp
#include "Agent.h"

void Agent::renew_position(p cur)
{
	this->cur = cur;
}

void Agent::set_path(Path path)
{
	this->path = path;
}

void Agent::set_astar_path(Map map, Map potential_map, string type)
{
	if (type == "renew")
		path = AStar(cur, goal, map, potential_map);
	else if (type == "init")
		path = AStar(start, goal, map, potential_map);
}

bool Agent::move_path()
{
	if (path.size() > 1)
	{
		cur = path[path.size() - 1];
		return true;
	}
	cur = path[0];
	return false;
}

void Agent::print_position()
{
	cout << "Agent " << this->name << ": ( " << this->cur.first << ", " << this->cur.second << " )" << endl;
}

void Agent::set_position()
{
	this->position.push_back({ this->cur, this->aTime });
}